pkg/client/linstor.go

```diff
@@ -1148,55 +1148,53 @@ func (s *Linstor) Mount(vol *volume.Info, source, target, fsType string, readonl
 		"blockAccessMode": block,
 	}).Info("mounting volume")
 
-	// Check if the path is a device
-	isDevice, err := s.mounter.PathIsDevice(source)
+	info, err := os.Stat(source)
 	if err != nil {
-		return fmt.Errorf("checking device path failed: %v", err)
-	}
-	if !isDevice {
-		return fmt.Errorf("%s does not appear to be a block device", source)
+		return fmt.Errorf("failed to stat source device: %w", err)
 	}
 
-	// Determine if we have exclusive access to the device. This is mostly
-	// a way to determine if a disklessly attached device's connection is down.
-	_, err = s.mounter.DeviceOpened(source)
-	if err != nil {
-		return fmt.Errorf("checking for exclusive open failed: %v, check device health", err)
+	if (info.Mode() & os.ModeDevice) != os.ModeDevice {
+		return fmt.Errorf("path %s is not a device", source) //nolint:goerr113
 	}
 
 	if readonly {
 		options = append(options, "ro")
 
 		// This requires DRBD 9.0.26+, older versions ignore this flag
-		err := s.setDevReadOnly(source)
+		err := s.setDevReadOnly(ctx, source)
 		if err != nil {
 			return fmt.Errorf("failed to set source device readonly: %w", err)
 		}
 	} else {
 		// We might be re-using an existing device that was set RO previously
-		err = s.setDevReadWrite(source)
+		err = s.setDevReadWrite(ctx, source)
 		if err != nil {
 			return fmt.Errorf("failed to set source device readwrite: %w", err)
 		}
 	}
 
 	// This is a regular filesystem so format the device and create the mountpoint.
 	if !block {
-		if err := s.formatDevice(vol, source, fsType); err != nil {
+		if err := s.formatDevice(ctx, vol, source, fsType); err != nil {
 			return fmt.Errorf("mounting volume failed: %v", err)
 		}
-		if err := s.mounter.MakeDir(target); err != nil {
+
+		if err := os.MkdirAll(target, os.FileMode(0755)); err != nil {
 			return fmt.Errorf("could not create target directory %s, %v", target, err)
 		}
 		// This is a block volume so create a file to bindmount to.
 	} else {
-		err := s.mounter.MakeFile(target)
-		if err != nil {
+		f, err := os.OpenFile(target, os.O_CREATE, os.FileMode(0644))
+		if err != nil && !os.IsExist(err) {
 			return fmt.Errorf("could not create bind target for block volume %s, %v", target, err)
 		}
+
+		if f != nil {
```

Comment:
hm, in general I don't like to rely on the "good value" of a function if it returned an error. Usually you can not and should not.

```go
if err != nil {
    if !os.IsExist(err) {...}
} else {
    _ = f.Close()
}
```